src/CompiledMatcher.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver;
+
+use Composer\Semver\Constraint\Constraint;
+use Composer\Semver\Constraint\ConstraintInterface;
+use Composer\Semver\Constraint\NotCompilableConstraintException;
+
+/**
+ * Helper class to evaluate constraint by compiling and reusing the code to evaluate
+ */
+class CompiledMatcher
+{
+    private static $checked = array();
```

Comment:
I would rename this property to `compiledCheckerCache` or something like that.

Reply:
thanks